From bb315839cd846ad23ee7c8e0a2bffe5d58d9252c Mon Sep 17 00:00:00 2001
From: Richard Poettler <richard.poettler@gmail.com>
Date: Wed, 29 Jul 2015 18:40:46 +0200
Subject: [PATCH] add /usr/share/pyfa to python searchpath

---
 pyfa.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pyfa.py b/pyfa.py
index 76b7ca3..8366f6d 100755
--- a/pyfa.py
+++ b/pyfa.py
@@ -19,6 +19,8 @@
 #===============================================================================
 
 import sys
+# we installed pyfa specific modules into another directory
+sys.path.append("/usr/share/pyfa")
 import re
 
 if not hasattr(sys, 'frozen'):
-- 
2.4.3

